Alerting: Support passing tags to Pagerduty and allow notification on specific event categories

johntdyer opened the pull request:
What this PR does / why we need it: 


* This PR add panel tags to the Pagerduty panel. Please note this is already supported in the Prometheus Alert Manager and webhook notifiers.( #10989 ). 
* This PR also allows one to define a notification channel w/ another severity, rather then just critical
* I also allow users to define `group`, `component` and `class` tags which are top level in the v2 api. 
* I also increased test coverage substantially for the component 

Which issue(s) this PR fixes: n/a

replaces PR #19912 / #19913

Fixes #19913

Special notes for your reviewer:

feature is already supported in Webhook and Prometheus notifiers. I am simply extending that support to the pagerduty one

johntdyer committed:
Support passing tags to pagerduty api, and allow notification on specific severity.  Also allow override of group, class, component at payload level for v2 eevent api

johntdyer commented:
@maintainers - So the CircleCI test seems to have failed because of the addition of jsonassert...   What are my options ?

papagian commented:
@johntdyer 
The tests are failing because the new dependency `jsonassert` should also be under the `vendor` directory. To do so you need to run `go mod vendor` but I'm not fond of the idea of introducing a new dependency only for the tests.

papagian commented on pkg/services/alerting/notifiers/pagerduty.go:
```diff
@@ -23,7 +24,7 @@ func init() {
         <span class="gf-form-label width-14">Integration Key</span>
         <input type="text" required class="gf-form-input max-width-22" ng-model="ctrl.model.settings.integrationKey" placeholder="Pagerduty Integration Key"></input>
-      </div>
+			</div>
```
This modification introduces wrong indentation.

johntdyer replied:
freaking tabs..... :)

papagian commented on pkg/services/alerting/notifiers/pagerduty.go:
```diff
@@ -79,13 +80,8 @@ type PagerdutyNotifier struct {
-		return nil
-	}
+// BuildEventPayload is responsible for building the event payload body for sending to Pagerduty v2 API
+func (pn *PagerdutyNotifier) BuildEventPayload(evalContext *alerting.EvalContext) ([]byte, error) {
```
`BuildEventPayload` is only used inside this package so there is no need to be exported.

papagian commented on pkg/services/alerting/notifiers/pagerduty.go:
```diff
@@ -99,6 +95,26 @@ func (pn *PagerdutyNotifier) Notify(evalContext *alerting.EvalContext) error {
+	for _, tag := range evalContext.Rule.AlertRuleTags {
+		customData.Set(tag.Key, tag.Value)
+
+		// Override tags appropritatly if the are in the Pagerduty v2 API
```
It should be `appropriately` instead.

papagian reviewed (dismissed):
Thank you for your contribution!
Please check my comments for some changes.

aknuds1 commented on pkg/services/alerting/notifiers/pagerduty.go:
```diff
@@ -99,6 +95,26 @@ func (pn *PagerdutyNotifier) Notify(evalContext *alerting.EvalContext) error {
+			payloadJSON.Set("group", tag.Value)
+		}
+
+		if strings.ToLower(tag.Key) == "class" {
```
Isn't it better to use the `else if` form in order to improve readability?

kylebrandt replied:
```
switch strings.ToLower(tag.Key) {
  case "group":
    ....
```

johntdyer replied:
yea, I wasnt familiar w/ the `fallthrough` operator in a go switch.... learn something new every day :)

aknuds1 commented on pkg/services/alerting/notifiers/pagerduty.go:
```diff
@@ -99,6 +95,26 @@ func (pn *PagerdutyNotifier) Notify(evalContext *alerting.EvalContext) error {
 	pn.log.Info("Notifying Pagerduty", "event_type", eventType)
 
 	payloadJSON := simplejson.New()
+	bodyJSON := simplejson.New()
```
I don't see the reason for moving the `bodyJSON` declaration here, since it's not used until later on in any case.

johntdyer replied:
not sure why I did that, so I'll move back

aknuds1 commented on pkg/services/alerting/notifiers/pagerduty.go:
```diff
@@ -143,6 +158,20 @@ func (pn *PagerdutyNotifier) Notify(evalContext *alerting.EvalContext) error {
+func (pn *PagerdutyNotifier) Notify(evalContext *alerting.EvalContext) error {
+	body, err := pn.BuildEventPayload(evalContext)
+	if err != nil {
+		pn.log.Error("Unable to build event pagerduty payload: ", err)
```
Please return the error here.

johntdyer replied:
ack

aknuds1 commented on pkg/services/alerting/notifiers/pagerduty.go:
```diff
@@ -143,6 +158,20 @@ func (pn *PagerdutyNotifier) Notify(evalContext *alerting.EvalContext) error {
+	if err != nil {
+		pn.log.Error("Unable to build event pagerduty payload: ", err)
+	}
+	if evalContext.Rule.State == models.AlertStateOK && !pn.AutoResolve {
```
Why not execute this before line 166, since `body` isn't used if this condition is true?

johntdyer replied:
good point

aknuds1 reviewed (changes requested):
I saw some issues.

johntdyer committed:
few changes from PR

johntdyer committed:
return error and small change to ordering of conditions in Notify entry point

johntdyer committed:
Indentation is far to hard ;-P

johntdyer committed:
Remove fallthrough and add jsonassert to vendor dir

johntdyer commented:
> Not sure you want fallthrough here? That makes it so if `tag.Key` were say "group" and tag.Value is "foo", "class" would get to the tag.Value of "foo" as well. https://github.com/golang/go/wiki/Switch#fall-through

Thanks, good catchg,,, guess I didnt understand it like I thought I did :)

johntdyer committed:
Add vendor dir

aknuds1 commented on pkg/services/alerting/notifiers/pagerduty.go:
```diff
@@ -32,7 +33,7 @@ func init() {
             ng-options="s for s in ['critical', 'error', 'warning', 'info']">
           </select>
-        </div>
+      </div>
```
```suggestion
        </div>
```

aknuds1 commented on pkg/services/alerting/notifiers/pagerduty.go:
```diff
@@ -100,6 +96,23 @@ func (pn *PagerdutyNotifier) Notify(evalContext *alerting.EvalContext) error {
+	for _, tag := range evalContext.Rule.AlertRuleTags {
+		customData.Set(tag.Key, tag.Value)
+
+		// Override tags appropriately if the are in the Pagerduty v2 API
```
```suggestion
		// Override tags appropriately if they are in the PagerDuty v2 API
```

aknuds1 commented on pkg/services/alerting/notifiers/pagerduty.go:
```diff
@@ -100,6 +96,23 @@ func (pn *PagerdutyNotifier) Notify(evalContext *alerting.EvalContext) error {
+	payloadJSON.Set("component", "Grafana")
+
+	for _, tag := range evalContext.Rule.AlertRuleTags {
+		customData.Set(tag.Key, tag.Value)
```
Do you want to set the same field both in customData and payloadJson if a tag is in the PD API? Asking because I'm unsure.

johntdyer replied:
@aknuds1  

I think so yes... 

IIRC customData shows up in the actual pagerduty event which is passes to thee client, like mobile..  

<img width="290" alt="image" src="https://user-images.githubusercontent.com/58234/72008366-a57d8e00-3221-11ea-8977-65633676a459.png">

Similar to thresholds

However adding them to the payload portion allows you to classify and filter your events using Pagerduty's tooling via the UI.  You can see in the v2 api `class`, `group` and `component` have special meaning.  However I also think seeing them in the client, ie mobile would be helpful, so i put them in both.. hope that makes sense

johntdyer replied:
@aknuds1 / @papagian  - With regards to jsonassert, I wasn't sure of another way to test the json results of those methods... I just went w/ the tool I knew.. If you feel strongly about not adding this dependency I certainly understand... I just hope you may have some input into a more idiomatic way to accomplish what I am after here :)

johntdyer replied:
we good on this one ?

aknuds1 replied:
Yeah I'll resolve it.

aknuds1 commented on pkg/services/alerting/notifiers/pagerduty.go:
```diff
@@ -111,10 +124,9 @@ func (pn *PagerdutyNotifier) Notify(evalContext *alerting.EvalContext) error {
 	payloadJSON.Set("custom_details", customData)
-
 	bodyJSON := simplejson.New()
+
```
```suggestion
```

aknuds1 commented on pkg/services/alerting/notifiers/pagerduty_test.go:
```diff
@@ -98,6 +101,126 @@ func TestPagerdutyNotifier(t *testing.T) {
+					"autoResolve": false
+				}`
+
+				settingsJSON, jerr := simplejson.NewJson([]byte(json))
```
```suggestion
				settingsJSON, err := simplejson.NewJson([]byte(json))
```

johntdyer replied:
fixed

aknuds1 commented on pkg/services/alerting/notifiers/pagerduty_test.go:
```diff
@@ -98,6 +101,126 @@ func TestPagerdutyNotifier(t *testing.T) {
+				}`
+
+				settingsJSON, jerr := simplejson.NewJson([]byte(json))
+				So(jerr, ShouldBeNil)
```
```suggestion
				So(err, ShouldBeNil)
```

aknuds1 commented on pkg/services/alerting/notifiers/pagerduty_test.go:
```diff
@@ -98,6 +101,126 @@ func TestPagerdutyNotifier(t *testing.T) {
+					Message: "someMessage",
+					State:   models.AlertStateAlerting,
+				})
+
```
```suggestion
```

aknuds1 commented on pkg/services/alerting/notifiers/pagerduty_test.go:
```diff
@@ -98,6 +101,126 @@ func TestPagerdutyNotifier(t *testing.T) {
+				So(err, ShouldBeNil)
+
+				ja := jsonassert.New(t)
+
```
```suggestion
```

aknuds1 commented on pkg/services/alerting/notifiers/pagerduty_test.go:
```diff
@@ -98,6 +101,126 @@ func TestPagerdutyNotifier(t *testing.T) {
+					"autoResolve": false
+				}`
+
+				settingsJSON, jerr := simplejson.NewJson([]byte(json))
```
```suggestion
				settingsJSON, err := simplejson.NewJson([]byte(json))
```

johntdyer replied:
fixed

aknuds1 commented on pkg/services/alerting/notifiers/pagerduty_test.go:
```diff
@@ -98,6 +101,126 @@ func TestPagerdutyNotifier(t *testing.T) {
+				}`
+
+				settingsJSON, jerr := simplejson.NewJson([]byte(json))
+				So(jerr, ShouldBeNil)
```
```suggestion
				So(err, ShouldBeNil)
```

aknuds1 commented on pkg/services/alerting/notifiers/pagerduty_test.go:
```diff
@@ -98,6 +101,126 @@ func TestPagerdutyNotifier(t *testing.T) {
+				So(err, ShouldBeNil)
+
+				ja := jsonassert.New(t)
+
```
```suggestion
```

aknuds1 commented on pkg/services/alerting/notifiers/pagerduty_test.go:
```diff
@@ -98,6 +101,126 @@ func TestPagerdutyNotifier(t *testing.T) {
+					"client_url":"",
+					"dedup_key":"alertId-0",
+					"event_action":"trigger",
+
```
```suggestion
```

aknuds1 reviewed (changes requested):
I made some suggestions. I'm also unsure if we want to use jsonassert @papagian?

johntdyer committed:
Update pkg/services/alerting/notifiers/pagerduty_test.go

Co-Authored-By: Arve Knudsen <arve.knudsen@gmail.com>

johntdyer committed:
Update pkg/services/alerting/notifiers/pagerduty_test.go

Co-Authored-By: Arve Knudsen <arve.knudsen@gmail.com>

johntdyer committed:
Update pkg/services/alerting/notifiers/pagerduty_test.go

Co-Authored-By: Arve Knudsen <arve.knudsen@gmail.com>

johntdyer committed:
Update pkg/services/alerting/notifiers/pagerduty.go

fix grammar

Co-Authored-By: Arve Knudsen <arve.knudsen@gmail.com>

johntdyer committed:
Update pkg/services/alerting/notifiers/pagerduty.go

indentation is still hard

Co-Authored-By: Arve Knudsen <arve.knudsen@gmail.com>

johntdyer committed:
Update pkg/services/alerting/notifiers/pagerduty_test.go

whitespace

Co-Authored-By: Arve Knudsen <arve.knudsen@gmail.com>

johntdyer committed:
Update pkg/services/alerting/notifiers/pagerduty.go

Co-Authored-By: Arve Knudsen <arve.knudsen@gmail.com>

johntdyer committed:
fix naming of err variable in tests per PR suggestion

johntdyer committed:
Update pkg/services/alerting/notifiers/pagerduty_test.go

remove whitespace

Co-Authored-By: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 commented on pkg/services/alerting/notifiers/pagerduty.go:
```diff
@@ -143,6 +155,23 @@ func (pn *PagerdutyNotifier) Notify(evalContext *alerting.EvalContext) error {
+
+	body, err := pn.buildEventPayload(evalContext)
+	if err != nil {
+		pn.log.Error("Unable to build event pagerduty payload: ", err)
```
```suggestion
		pn.log.Error("Unable to build event PagerDuty payload", "error", err)
```

johntdyer replied:
ah, thank you... fixed and pushed

aknuds1 reviewed (changes requested):
I noticed there's an error in the logging call.

johntdyer committed:
Fix error logging for failing to build event

johntdyer reviewed (commented):
fixed

aknuds1 reviewed (commented):
Looks pretty good to me, but we have to decide if we want to introduce jsonassert or not as a dependency.

johntdyer commented:
@aknuds1  - Gotcha... well as I stated earlier, I am happy to go about the test another way..  I just went w/ the tool I was most familiar w/ to solve the problem at tjhe time...   I'll be standing by for your update.   Thanks again for all your help

aknuds1 commented:
@johntdyer No worries, I think I will try a solution without jsonassert, when I have the time.

marefr commented:
This new functionality needs documentation update here https://grafana.com/docs/grafana/latest/alerting/notifications/#pagerduty

aknuds1 commented:
> This new functionality needs documentation update here https://grafana.com/docs/grafana/latest/alerting/notifications/#pagerduty

Thanks @marefr, making a note of it.

johntdyer committed:
Fix docs for pagerduty handlers support of alert tags

johntdyer commented:
@aknuds1  / @marefr  - I fixed the table on notifications.md for the alert tags support for this channel..  Do we need anything else to get this merged ?

aknuds1 commented:
@johntdyer I will have a look again now, thanks.

aknuds1 committed:
Merge branch 'master' into pagerduty-add-panel-tags-rebase

mckn committed:
Fix: when clicking a plot on a touch device we won't display the annotation menu (#21479)

* Fixed so when clicking a plot on a touch device we won't display the annotation context menue.

* Refactorings after feedback.

* changed to standard-ish comment.

aknuds1 committed:
Drop jsonassert

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Fix up vendor/

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 reviewed (approved):
LGTM

aknuds1 commented:
@marefr Can you please let us know if there's more that needs to be added to the docs?

robbymilo committed:
Docs: add test for website build (#21364)

* add build-docs-website job to circleci

* update commands

* update readme command

* remove container when test finishes

* move build-docs-website to build-branches-and-prs

* update readme

aknuds1 committed:
ImportDashboardCommand: Validate JSON fields (#21350)

* ImportDashboardCommand: Validate JSON fields

marefr committed:
Chore: Adds basic alerting notification service tests (#21467)

Adds basic alerting notification tests. First steps towards 
getting #18011 merged.

Ref #18011

marefr committed:
Add disabled option for cookie samesite attribute (#21472)

Breaking change: If disabled the cookie samesite cookie attribute
will not be set, but if none the attribute will be set and is a
breaking change compared to before where none did not render the
attribute. This was due to a known issue in Safari.

Co-Authored-By: Arve Knudsen <arve.knudsen@gmail.com>
Co-Authored-By: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

Fixes #19847

ivanahuckova committed:
Explore: Fix timepicker when browsing back after switching datasource (#21454)

ryantxu committed:
Templating: use default datasource when missing (#21495)

dprokop committed:
DataLinks: Make data links input grow again (#21499)

marefr commented:
@aknuds1 A few lines below the table here https://grafana.com/docs/grafana/latest/alerting/notifications/#pagerduty regarding tags and group, component and class would be good. Otherwise this feature will only be known by the author of this PR basically.

torkelo committed:
SideMenu: Fixes issue with logout link opened in new tab (#21488)

* SideMenu: Fixes issue with logout link opened in new tab

* Updated snapshot

marefr committed:
Fix importing plugin dashboards (#21501)

#21350 introduced a bug regarding import of plugin dashboards. 
This should fix this and add custom validation if not importing 
plugin dashboard and dashboard property is missing.

Ref #21350

Co-Authored-By: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 commented:
> @aknuds1 A few lines below the table here https://grafana.com/docs/grafana/latest/alerting/notifications/#pagerduty regarding tags and group, component and class would be good. Otherwise this feature will only be known by the author of this PR basically.

@johntdyer Do you mind writing up this documentation as outlined above?

peterholmberg committed:
Tabs: Hide Tabs on Page header on small screens (#21489)

* Adding className to hide tabs on small screens

* update snapshot

* Readd border on pageheader

* update snap

woodsaj committed:
Auth: Rotate auth tokens at the end of requests (#21347)

By rotating the auth tokens at the end of the request we ensure
that there is minimum delay between a new token being generated
and the client receiving it.
Adds auth token slow load test which uses random latency for all 
tsdb queries..
Cleans up datasource proxy response handling.
DefaultHandler in middleware tests should write a response, the 
responseWriter BeforeFuncs wont get executed unless a response
is written.

Fixes #18644 

Co-authored-by: Marcus Efraimsson <marcus.efraimsson@gmail.com>

marefr committed:
Backend Plugins: Collect and expose metrics and plugin process health check (#21481)

Adds support for collecting metrics from backend plugins and 
exposing them thru Grafana's Prometheus metrics endpoint. 
Enables to check health of backend plugin by using the route
`/api/plugins/<plugin id>/health`.
Uses sdk v0.6.0.

Closes #20984

aocenas committed:
Add changelog for 6.5.3

sakjur committed:
Admin: Add promotional page for Grafana Enterprise (#21422)


Co-authored-by: Torkel Ödegaard <torkel@grafana.com>

sunker committed:
CloudWatch: Annotation Editor rewrite (#20765)

* wip: react rewrite

* Cleanup

* Break out non annontations specific fields

* Cleanup. Make annontations editor a functional component

* Remove redundant classnames

* Add paneldata to props

* Cleanup

* Fix rebase merge problem

* Updates after pr feedback

* Fix conflict with master

ivanahuckova committed:
Fix regex in convertCSSToStyle, add test coverage (#21508)

dprokop committed:
grafana/data: Update plugin config page typings (BREAKING) (#21503)

* Change typyings of plugins config pages

* Update plugin config page types of core plugins

ryantxu committed:
DataLinks: allow using values from other fields in the same row (#21478)

ryantxu committed:
DataFrames: add arrow test and capture metadata parsing errors (#21524)

stevenvachon committed:
@grafana/toolkit: cleanup (#21441)

* Clarify readme for extended webpack config

* Add support for ES2019 features

* Build task: path.resolve → resolvePath

* Build task: avoid fs race conditions

* Build task: use async fs functions

* Build task: use rimraf directly

... because depending on the workspace parent to have it installed is fragile, and a child process is always slower.

* Build task: misc

ryantxu committed:
AppPlugin: remove simple app from the core repo (#21526)

lunaticusgreen committed:
Toolkit: support less loader (#21527)

dprokop committed:
Changelog generation: Generate grafana/ui changelog (#21531)

peterholmberg committed:
Inspect: Use AutoSizer for managing width for content in tabs. (#21511)

* wrap content in autosizer

* replace lib

* removing react-virtualized, correct type in editor.ts

* remove caret

* remove dep from grafana/ui and remove story util

kaydelaney committed:
Explore: Ensures queries aren't updated when returning to dashboard if browser back is used (#20897)

* Explore: Ensures queries aren't updated when returning to dashboard if browser back is used
Closes #20873

aocenas committed:
Update latest to 6.5.3 (#21509)

sakjur committed:
Build: Ignore content of /pkg/extensions, not directory (#21540)

papagian committed:
Fix TypeScript error (#21545)

kaydelaney committed:
Explore/Loki: Filter expression only treated as regex when regex operator is used (#21538)

* Explore/Loki: Filter expression only treated as regex when regex operator is used
Closes #17963

* Adds test cases to verify new behavior

alexanderzobnin committed:
fix dateMath import in grafana-ui (#21546)

kylebrandt committed:
Vendor: grafana-plugin-sdk-go v0.11.0 (#21552)

upgrades https://github.com/grafana/grafana-plugin-sdk-go from v0.6.0 to v0.11.0

ryantxu committed:
Templating: update variables on location changed (#21480)

oddlittlebird committed:
Docs: Update mac.md (#20782)

* Update mac.md

Reviewed and edited macOS install instructions.

* Updated mac.md and configuration.md

Removed developer build instructions from mac.md and moved configuration information from mac.md to configuration.md.

* Update mac.md

Updated to reflect the new fields on the Downloads page

* Update mac.md

* Update mac.md

oddlittlebird committed:
Docs: Update rpm.md (#21547)

Minor format fix

torkelo committed:
Dashboards: Default Home Dashboard Update (#21534)

* HomeLinks: WIP panel for getting started links

* Progress

* progress

* Updated styling

* Updated default home dashboard

* Update light card styles

* Tweaks based on feedback

sunker committed:
CloudWatch: Display partial result in graph when max DP/call limit is reached  (#21533)

* Return data and warning in case dp limit is reached

* Change warning text

Commit:
Add component: Cascader (#21410)

* Rename old cascader

* Change name of old cascader

* Add basic cascader without search

* Add basic cascader without search

* Flatten options to make it searchable

* Add regex search and make backspace work

* Add barebone search without styles

* Add SearchResult list

* Add search navigation

* Rewrite of cascader and add some things to SelectBase

* Make SelectBase controlllable

* Cleanup

* Add initial value functionality

* Add onblur to hand caret direction

* New storyboom format for ButtonCascader

* Add knobs to story

* Add story and docs for UnitPicker

* Make UnitPicker use Cascader

* Fix backspace issue and empty value

* Fix backspace issue for real

* Remove unused code

* Fix focus issue

* Change children to items and remove ButtonCascaderProps

* Remove local CascaderOption

* Fix failed test

* Revert UnitPicker changes and change format for ButtonCascader

* Fix failing tests

xlson committed:
CI: Enterprise MSI (#21518)

* build: builds msi for enterprise

* build: renamed msi generator directory

* build: build and deploy enterprise msi

* Build: placeholder license file for enterprise

* Build: correct name for ee msi

xlson committed:
CI: adds missing files for ee msi (#21559)

marefr committed:
Alerting: Fix image rendering and uploading timeout preventing to send alert notifications (#21536)

* svc alerting - use a shorter ctx to upload the img
This will prevent timeout on img upload to cancel the notifications from being sent

* components img uploader - pass the ctx to aws lib

* make webdavuploader use the ctx

* make azureblobuploader use the ctx

* rename uploadImage() to renderAndUploadImage()
for better clarity about what this method work

* Use timeout + 2s for plugin renderer (same as service and phantomjs)

* Make sure that original EvalContext is updated after render and upload

* Verify notification sent even if render or image upload times out

* fix lint

* fixes after review

Co-authored-by: Edouard Hur <3418467+hekmon@users.noreply.github.com>

Fixes #21018

xlson committed:
CI: Revert msi build (#21561)

* Revert "CI: adds missing files for ee msi (#21559)"

This reverts commit 6e412d88c9faf54197bd6d0e733e9f02e6ca386f.

* Revert "CI: Enterprise MSI (#21518)"

This reverts commit d910f69b2c68eed232f4a9ec131e4fa9305ae4ea.

sunker committed:
Stackdriver: Support meta labels (#21373)

* Rewrite angular segments for filter and group by in react

* wip: refactoring

* Update metric find queries

* Remove old maps used to create labels - use one map for all types instead

* Use value as label (again) for filters ang groupby

* Remove old filter

* Remove not used code

* Fixes after pr feedback

* Fix broken tests and add new metadata tests

* Add index file to make imports cleaner

* Cleanup. Remove old angular filter code

* Fix broken tests

* Use type switching instead of if statements

* Use globals for regex

* Updates after pr feedback

* Make sure it's possible to filter using the same key multiple times

* Replace metric select with segment component

* Pass template vars as props

* Refactor meta labels code

* Reorder template variables

* Fix broken tests

* Reset metric value when changing service

* Fix lint issue.

* Make tests independant of element order

* Include kubernetes.io in regex

* Add instruction in help section

stevenvachon committed:
Babel: use babel-loader instead of ts-loader, ng-annotate with babel-plugin-angularjs-annotate (#21554)

* Applied prettier to some webpack configs

* Removed ng-annotate

… and used same annotation approach as webpack.hot.js

* Removed redundant import

… that is problematic with Babel's module resolver

* Updated lockfile

* Replace ts-loader with babel-loader in webpack.dev

* Change tslint-loade order in dev webpack config

Co-authored-by: Dominik Prokop <dominik.prokop@grafana.com>

sunker committed:
CloudWatch: Multi-valued template variable dimension alias fix (#21541)

* Match dimension value with label substring

* Add unit tests

peterholmberg committed:
Inspect: Download DataFrame to Csv (#21549)

* basic setup for csv export

* tinkering with flex-grow

* fixed layout

* align select and button

* using stylesfactory

sunker committed:
CloudWatch: Calculate period based on time range (#21471)

* Calculate min period based on time range and no of queries

* Use hardcoded array of periods if period is not defined actively by the user

* Fix broken tests

* Use a smaller max period for auto interval

* Fix broken tests

* Test period calculation

* Test min retention period

* Fix broken test

marefr committed:
Provisioning: Start provision dashboards after Grafana server have started (#21564)

6.2-beta1 changed so that data sources, dashboards and alert notifiers 
are provisioned on service Init where's before only data sources and 
alert notifiers was provisioned on service Init and dashboards was first
provisioned in service Run which happens after Grafana server have 
started.
This change reverts so that dashboard provisioning are first happening 
in service Run, after Grafana server have started.

Fixes #21133

sunker committed:
Chore: Fix go vet problem (#21568)

ivanahuckova committed:
Explore: Create unique ids and deduplicate Loki logs (#21493)

papagian committed:
Docs: What's new in Grafana v6.6 Draft (#21562)

* Docs: What's new in Grafana v6.6 Draft

marefr committed:
CloudWatch: Fix ordering of map to resolve flaky test (#21572)

sunker committed:
Fix merge problem (#21574)

sakjur committed:
Config: Use license info instead of build info for feature toggling (#21558)

ivanahuckova committed:
Explore: Context tooltip to copy labels and values from graph (#21405)

dprokop committed:
Revert "Babel: use babel-loader instead of ts-loader, ng-annotate with babel-plugin-angularjs-annotate (#21554)" (#21570)

This reverts #21554 which caused problems with e2e.
This reverts commit e7e0d18bc85928e882030262dbdceb5771194844.

torkelo committed:
Minor style changes on upgrade page (#21566)

torkelo committed:
Redux: Fixed function adding a new reducer (#21575)

marefr committed:
CloudWatch: Fix ordering of map to resolve flaky test take 2 (#21577)

Commit:
ButtonCascader: Fix error in Explore (#21585)

* Add children and remove nullified icon

* Revert data format with children

* Update test

* Revert styles

* Update types

seanson committed:
OAuth: Fix role mapping from id token (#20300)

As part of the new improvements to JMESPath in #17149 a 
commit ensuring Role and Email data could be queried from 
id_token was lost.
This refactors and fixes extracting from both token and user 
info api consistently where before only either token or 
either user info api was used for extracting data/attributes.

Fixes #20243

Co-authored-by: Timo Wendt <timo@tjwendt.de>
Co-authored-by: twendt <timo@tjwendt.de>
Co-authored-by: henninge <henning@eggers.name>
Co-Authored-by: Henning Eggers <henning.eggers@inovex.de>
Co-authored-by: Marcus Efraimsson <marcus.efraimsson@gmail.com>

peterholmberg committed:
Inspect: table take full height in drawer (#21580)

torkelo committed:
Typescript: null check fixes, and news panel fix (#21595)

marefr committed:
Render: Use https as protocol when rendering if HTTP2 enabled (#21600)

This fixes so that protocol is https instead of h2 when rendering
image if Grafana is configured to use HTTP2.

Ref grafana/grafana-image-renderer#77

ryantxu committed:
Toolkit: fix prettier error reporting (#21599)

dprokop committed:
E2E docs: Add guide to debuging PhantomJS (#21606)

* Add guide to debuging PhanomJS

* Update contribute/style-guides/e2e.md

xlson committed:
CI: MSI for Enterprise (#21569)

* CI: ee msi build

* shellcheck

* Fake it so that we get a ee msi build on pull request runs

* fix error in prep job

* save msi for testing

* get enterprise zip

* Fixes to version and hash detection.

* Removes ci test code

xlson committed:
Build: enables deployment of enterprise msi (#21607)

ivanahuckova committed:
Explore: Remove destructuring of empty state in LogRowMessage (#21579)

ivanahuckova committed:
grafana/ui: Add synced timepickers styling to TimePicker (#21598)

daniellee committed:
graphite: does not modify last segment when... (#21588)

opening the Graphite query editor and there is no data
for the last segment value.

Fixes #21563

torkelo committed:
TestData: Adds important new features to the random walk scenario (#21613)

* test data refactor

* new random walk options

* fixed test

* Added automatic names

torkelo committed:
BigValue: Updated test dashboard and made some chart sizing tweaks (#21616)

xlson committed:
Build: adds missing filters required to build oss msi (#21618)

marefr committed:
Devenv: InfluxDB logs dashboard (#21620)

Test dashboard for using influxdb with logs panel

sunker committed:
CloudWatch: Expand dimension value in alias correctly (#21626)

* Make sure dimension value is being returned, and not just label

* Fix typo

papagian committed:
Update musl checksums (#21621)

ryantxu committed:
TestData: allow negative values for random_walk parameters (#21627)

Commit:
Emotion: Add main package with version 10 (#21560)

* Update version and snapshots

Co-authored-by: Dominik Prokop <dominik.prokop@grafana.com>

kaydelaney committed:
Chore: Remove angular dependency from backendSrv (#20999)

* Chore: Remove angular dependency from backendSrv

* Refactor: Naive soultion for logging out unauthorized users

* Refactor: Restructures to different streams

* Refactor: Restructures datasourceRequest

* Refactor: Flipped back if statement

* Refactor: Extracted getFromFetchStream

* Refactor: Extracts toFailureStream operation

* Refactor: Fixes issue when options.params contains arrays

* Refactor: Fixes broken test (but we need a lot more)

* Refactor: Adds explaining comments

* Refactor: Adds latest RxJs version so cancellations work

* Refactor: Cleans up the takeUntil code

* Refactor: Adds tests for request function

* Refactor: Separates into smaller functions

* Refactor: Adds last error tests

* Started to changed so we require getBackendSrv from the @grafana-runtime when applicable.

* Using the getBackendSrv from @grafana/runtime.

* Changed so we use the getBackendSrv from the @grafana-runtime when possible.

* Fixed so Server Admin -> Orgs works again.

* Removed unused dependency.

* Fixed digest issues on the Server Admin -> Users page.

* Fix: Fixes digest problems in Playlists

* Fix: Fixes digest issues in VersionHistory

* Tests: Fixes broken tests

* Fix: Fixes digest issues in Alerting => Notification channels

* Fixed digest issues on the Intive page.

* Fixed so we run digest after password reset email sent.

* Fixed digest issue when trying to sign up account.

* Fixed so the Server Admin -> Edit Org works with backendSrv

* Fixed so Server Admin -> Users works with backend srv.

* Fixed digest issues in Server Admin -> Orgs

* Fix: Fixes digest issues in DashList plugin

* Fixed digest issues on Server Admin -> users.

* Fix: Fixes digest issues with Snapshots

* Fixed digest issue when deleting a user.

* Fix: Fixes digest issues with dashLink

* Chore: Changes RxJs version to 6.5.4 which includes the same cancellation fix

* Fix: Fixes digest issue when toggling folder in manage dashboards

* Fix: Fixes bug in executeInOrder

* Fix: Fixes digest issue with CreateFolderCtrl and FolderDashboardsCtrl

* Fix: Fixes tslint error in test

* Refactor: Changes default behaviour for emitted messages as before migration

* Fix: Fixes various digest issues when saving, starring or deleting dashboards

* Fix: Fixes digest issues with FolderPickerCtrl

* Fixed digest issue.

* Fixed digest issues.

* Fixed issues with angular digest.

* Removed the this.digest pattern.

Co-authored-by: Hugo Häggmark <hugo.haggmark@gmail.com>
Co-authored-by: Marcus Andersson <systemvetaren@gmail.com>

xlson committed:
Changelog: v6.6.0-beta1 (#21619)

* Changelod: v6.6.0-beta1

* Changelog: grafana/ui v6.6.0-beta.1.0

* Changelog: moved toolkit to its own file

* Changelog: fix version

* Remove closed PR

Co-authored-by: Sofia Papagiannaki <papagian@users.noreply.github.com>

ceh committed:
Plugins: Apply adhoc filter in Elasticsearch logs query (#21346)

Fixes #21086

torkelo committed:
Docs: Updated what's new article (#21624)

* Docs: Updated what's new articel

* fixed spelling

* Updated

torkelo committed:
Docs: Update changelog with attribution (#21637)

xlson committed:
latest.jso: Update latest beta 6.6.0-beta1 (#21623)

xlson committed:
next version 6.7.0 (#21617)

* next version 6.7.0

* updates grafana/toolkit and co

* Verison: manually fixed version

kenju committed:
CloudWatch: Add DynamoDB Accelerator (DAX) metrics & dimensions (#21644)

Closes #10494

torkelo committed:
Templating: A way to support object syntax for global vars (#21634)

ivanahuckova committed:
Explore: Fix log level color and add tests (#21646)

johntdyer committed:
Fix golint error w/ variable naming

johntdyer committed:
Add docs for taging

johntdyer committed:
Merge branch 'master' into pagerduty-add-panel-tags-rebase

johntdyer commented:
@oddlittlebird - Added

johntdyer committed:
Spelling is hard

oddlittlebird commented on docs/sources/alerting/notifications.md:
```diff
@@ -103,6 +103,10 @@ Integration Key | Integration key for PagerDuty.
 
+#### Note:
+
+The tags `Class`, `Group`, and `Component` have special meaning in the [Pagerduty Common Event Format - PD-CEF](https://support.pagerduty.com/docs/pd-cef).  If an alert panel defines these tag keys they will be transposed to the root of the event sent to Pagerduty.  This means they will be avaialble within the Pagerduty UI and Filtering tools.
```
Please remove the double spaces and fix the type in the third sentence. "available"

"Note:" should not be a heading, it should be bold and in-line with the paragraph

johntdyer replied:
Fixed, @oddlittlebird  please review

oddlittlebird reviewed (changes requested):
I only saw changes to notifications.md. Are there any other doc changes I should be aware of?

johntdyer committed:
Fix heading for docs

johntdyer commented:
@oddlittlebird 

> I only saw changes to notifications.md. Are there any other doc changes I should be aware of?

None that I am aware of...

marefr reviewed (approved):
LGTM

marefr commented:
@aknuds1 think we should be ready to merge this and include in 6.6 stable. What do you think? @oddlittlebird can you approve since merging is currently blocked?

oddlittlebird reviewed (approved):
Approved

aknuds1 committed:
alerting: Fix indentation in pagerduty test

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 reviewed (approved):
LGTM!

aknuds1 commented:
Looks good to me @marefr.

marefr committed:
Merge remote-tracking branch 'origin/master' into pagerduty-add-panel-tags-rebase

marefr commented:
@johntdyer Thank you for contributing to Grafana!

marefr commented:
This will be included in Grafana v6.7 release and not v6.6 since it includes changes to dependencies and such and we're really close to releasing 6.6 stable.

johntdyer commented:
@marefr  how do I get the change mentioned on the changelog?

papagian commented:
> @marefr how do I get the change mentioned on the changelog?

@johntdyer
During the 6.7 release process we will run a script that generates the Changelog by selecting the merged PRs labeled as `add to changelog`.
